docs/protocol-tutorial/0010-states-channels.md

```diff
@@ -0,0 +1,305 @@
+
+### ChallengeDuration
+
+This is duration (in seconds) of the challenge-response window. If a challenge is raised on chain at time `t`, the channel will finalize at `t + ChallengeDuration` unless cleared by a subqsequent on-chain transaction.
+
+!!! warning
+
+    This should be at least 1 block time (~15 seconds on mainnet) and less than `2^48-1` seconds. Whatever it is set to, the channel should be closed long before `2^48 - 1 - challengeDuration`. In practice we recommend somewhere between 5 minutes and 5 months.
+
+### AppData
+
+The AppData is optional data which may be interpreted by the Nitro application and affect the execution rules of the channel -- see the section on [application rules](./0020-execution-rules.md#application-rules). For example, it could describe the state of a chess board or include the hash of a secret.
+
+### Outcome
+
+This describes how funds will be disbursed if the channel were to finalize in the current state. See the section on [Outcomes](./0030-outcomes.md).
+
+### TurnNum
+
+The turn number is the mechanism by which newer states take precedence over older ones. The turn number usually increments as the channel progresses.
+
+!!! warning
+
+    The turn number must not exceed 281,474,976,710,655 because then it will overflow on chain. It should not exceed 4,294,967,295 because it may then overflow off-chain. It is very unlikely a channel would ever have this many updates.
+
+### IsFinal
+
+This is a boolean flag which allows the [channel execution rules](#execution-rules) to be bypassed and for the channel to be finalized "instantly" without waiting for the challenge-response window to lapse.
+
+!!! warning
+
+    As soon as an `isFinal=true` state is _enabled_ (that is to say, you cannot prevent it from becoming supported) it is not safe to continue executing the state channel. It should be finalized immediately.
+
+## Channel IDs
+
+Channels are identified by the hash of the `FixedPart`` of the state (those parts that may _not_ vary):
+
+```solidity
+
+  bytes32 channelId = keccak256(
+      abi.encode(
+          fixedPart.chainId,
+          fixedPart.participants,
+          fixedPart.channelNonce,
+          fixedPart.appDefinition,
+          fixedPart.challengeDuration
+      )
+  );
+
+```
+
+## State commitments
+
+To commit to a state, a hash is formed as follows:
+
+```solidity
+ bytes32 stateHash = keccak256(abi.encode(
+        channelId,
+        variablePart.appData,
+        variablePart.outcome,
+        variablePart.turnNum,
+        variablePart.isFinal
+    ));
+```
+
+and this hash is signed using an _ephemeral_ Ethereum private key. _Ephemeral_ in this context means a dedicated private key, generated solely for the purpose of executing the state channel. The signature has the following type:
```

Comment:
I am not aware of any ephemeral keygen stuff happening in go-nitro. I even wrote a [notion doc](https://www.notion.so/statechannels/Hub-Hot-Wallet-Risk-ed5ed921c2fe4c03aac70c0966e3e825) about it!

Am I misunderstanding something here?

Reply:
This might be describing the way things ought to be more than the way things are. 

Right now we have injected dependencies, so I believe it is possible to instantiate a client where the same key is used by the chain service and by the store/protocols for signing state updates. But at least the current system allows for these to be different (granted, it does not enforce it). But technically we are outsourcing the responsibility for it to be ephemeral, and we should probably bring that in house eventually. And yes your notion doc has some ideas which we might want to use. 

Does that clarify things?

Reply:
Medium clarity. I'm not sure how close go-nitro is to supporting ephemeral keys. Currently we take a chainService and a store as injections. Store has interface methods 
```go
GetChannelSecretKey() *[]byte // Get a pointer to a secret key for signing channel updates
GetAddress() *types.Address   // Get the (Ethereum) address associated with the ChannelSecretKey
```
which don't seem well situated for handling a key / address per channel.

The supplied chain service can be applying a different key to transactions, but the protocols lean on the store's reported `myAddress` in order to fill out values in the participants list. Does NitroAdjudicator make any inferences based on the transaction's `msg.sender`? (I think not?) If it does, the transaction signer would have to match the address listed in `participants`.

An aside: If we design with the expectation of ephemeral keys per channel, we probably don't require nonces, because ephemeral keys map to ephemeral addresses (burner address), and there is a requirement for state signatures to be from the addresses of `Participants`.

Overall I think this is too far removed from current implementation to be included in the docs... ?

Reply:
> Does NitroAdjudicator make any inferences based on the transaction's msg.sender? (I think not?)

Correct, it does not.

>  don't seem well situated for handling a key / address per channel.

Yes we are far from supporting a key per channel -- and this is probably the best definition of what an "ephemeral key" ought to be. I've removed "ephemeral" from the docs for now [a715bb3](https://github.com/statechannels/go-nitro/pull/872/commits/a715bb31dfdb1b97fc61ef056d8572a459ab2850).

I think what i) we have typically meant by using "ephemeral" and ii) what is actually the most important thing from a security point of view is simply using a special key for channel updates which does not directly control any funds on chain. In hindsight this may not be a good usage of the term, as the key may actually end up being long-lived. 

See also https://github.com/statechannels/go-nitro/blob/502b7fe7bb01a71e0e1c8b07a5a3ef40a6f2eaa3/.adr/0002-signature-scheme.md#L13 where we could also remove the word "ephemeral".